### PR TITLE
services: Add Youdao OCR API

### PR DESCRIPTION
**Pull request by WhiredPlanck**

### CODE REVIEW

**Comment by WhiredPlanck** [comment]
测试结果似乎不错。但是才注意到有道这个 API 似乎没有什么免费额度…… 要不要 Close 算了……

File: fastocr/service.py [hunk 1]
```diff
@@ -53,9 +56,62 @@ async def basic_general(self, image: bytes, lang=''):
     async def close(self):
         await self.session.close()
 
+class YoudaoOcr(BaseOcr):
+    API_BASE = 'https://openapi.youdao.com/ocrapi'
+    CURTIME = str(int(time))
```
Comment (BruceZhang1993) [inline fastocr/service.py]: Maybe `str(time())` ?

File: fastocr/service.py [hunk 2]
```diff
@@ -53,9 +56,62 @@ async def basic_general(self, image: bytes, lang=''):
...
+    def __init__(self, setting: Setting):
+        self._sign = ''
+        self.appid = setting.get('YoudaoOCR', 'app_id') # appKey in Youdao docs
+        self.seckey = setting.get('YoudaoOCR', 'secret_key') # appSecret in Youdao docs
+        self.session = ClientSession()
+
+    @property
+    def sign(self):
+        if self._sign == '':
+            sign, _ = self.get_sign()
```
Comment (BruceZhang1993) [inline fastocr/service.py]: get_sign needs one param type: bytes

File: fastocr/service.py [hunk 3]
```diff
@@ -53,9 +56,62 @@ async def basic_general(self, image: bytes, lang=''):
...
+            'img': b64encode(image).decode(),
+            'langType': lang if lang != '' else 'auto',
+            'detectType': '10012',
+            'imageType': '1',
+            'appKey': self.appid,
+            'salt': self.SALT,
+            'sign': self.sign,
+            'docType': 'json',
+            'signType': 'v3',
+            'curtime': self.CURTIME
```
Comment (BruceZhang1993) [inline fastocr/service.py]: this class will always be singleton when this application running
`CURTIME` needs to update first

**Review by BruceZhang1993 — CHANGES_REQUESTED** [review]
request changes

File: fastocr/service.py (hunk 1 above)
Comment (WhiredPlanck) [reply fastocr/service.py]: Youdao use `int()` method in their demos, the test results seems to be OK and I have no idea that if `int()` is necessary. Anyway, I will remove it.

Comment (whriedplanck) [reply fastocr/service.py:61]: Youdao use `int()` method in their demos, the test results seems to be OK and I have no idea that if `int()` is necessary. Anyway, I will remove it.

**Comment by WhiredPlanck** [comment]
> request changes

Done.

**Review by BruceZhang1993 — APPROVED** [review]

**Pull request merged**